flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/table/MongoDBConnectorDeserializationSchema.java

```diff
@@ -0,0 +1,756 @@
+        if (docObj.isTimestamp()) {
+            Instant instant = convertToInstant(docObj.asTimestamp());
+            return (int) convertInstantToLocalDate(instant).toEpochDay();
+        }
+        throw new IllegalArgumentException(
+                "Unable to convert to date from unexpected value '"
+                        + docObj
+                        + "' of type "
+                        + docObj.getBsonType());
+    }
+
+    /**
+     * A conversion from TimeType to int describes the number of milliseconds of the day.
+     *
+     * @see org.apache.flink.table.types.logical.TimeType
+     */
+    private int convertToTime(BsonValue docObj) {
+        if (docObj.isDateTime()) {
+            Instant instant = convertToInstant(docObj.asDateTime());
+            return convertInstantToLocalTime(instant).toSecondOfDay() * 1000;
+        }
+        if (docObj.isTimestamp()) {
+            Instant instant = convertToInstant(docObj.asTimestamp());
+            return convertInstantToLocalTime(instant).toSecondOfDay() * 1000;
+        }
+        throw new IllegalArgumentException(
+                "Unable to convert to time from unexpected value '"
+                        + docObj
+                        + "' of type "
+                        + docObj.getBsonType());
+    }
+
+    private TimestampData convertToTimestamp(BsonValue docObj) {
+        if (docObj.isDateTime()) {
+            return TimestampData.fromLocalDateTime(
+                    convertInstantToLocalDateTime(convertToInstant(docObj.asDateTime())));
+        }
+        if (docObj.isTimestamp()) {
+            return TimestampData.fromLocalDateTime(
+                    convertInstantToLocalDateTime(convertToInstant(docObj.asTimestamp())));
+        }
+
+        throw new IllegalArgumentException(
+                "Unable to convert to timestamp from unexpected value '"
+                        + docObj
+                        + "' of type "
+                        + docObj.getBsonType());
+    }
+
+    private TimestampData convertToLocalTimeZoneTimestamp(BsonValue docObj) {
+        if (docObj.isDateTime()) {
+            return TimestampData.fromEpochMillis(docObj.asDateTime().getValue());
+        }
+        if (docObj.isTimestamp()) {
+            return TimestampData.fromEpochMillis(docObj.asTimestamp().getTime() * 1000L);
+        }
+        throw new IllegalArgumentException(
+                "Unable to convert to timestamp with local timezone from unexpected value '"
+                        + docObj
+                        + "' of type "
+                        + docObj.getBsonType());
+    }
+
+    private byte[] convertToBinary(BsonValue docObj) {
+        if (docObj.isBinary()) {
+            return docObj.asBinary().getData();
+        }
+        throw new UnsupportedOperationException(
+                "Unsupported BYTES value type: " + docObj.getClass().getSimpleName());
+    }
+
+    private StringData convertToString(BsonValue docObj) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.